### ISSUE TITLE
reconnect 

### ISSUE THREAD

**Issue by tgssb** [Rhymen/go-whatsapp#74]
how to check if is not connected, trying to reconnect

example

if( !isconnected){
login (wac)
}

**Comment by leandrocunha01** [comment]
`func wp() {
	wac, err := whatsapp.NewConn(5 * time.Second)
	wac.SetClientName("Browser X", "Linux")
	if err != nil {
		fmt.Fprintf(os.Stderr, "error creating connection: %v\n", err)
		return
	}
	wac.AddHandler(&middle.WpaHandler{})
	err = middle.Makelogin(wac)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Tentendo novamente...: %v\n", err)
		wp()
		//return
	}
}`

**Comment by leandrocunha01** [comment]
func wp() {
	wac, err := whatsapp.NewConn(5 * time.Second)
	wac.SetClientName("Browser X", "Linux")
	if err != nil {
		fmt.Fprintf(os.Stderr, "error creating connection: %v\n", err)
		return
	}
	wac.AddHandler(&middle.WpaHandler{})
	err = middle.Makelogin(wac)
	if err != nil {
		fmt.Fprintf(os.Stderr, "Tentendo novamente...: %v\n", err)
		wp()
		//return
	}
}

**Comment by Rhymen** [comment]
What are you trying to check? If the WebSocket disconnected after your initial login? Or if your mobile phone disconnected? This seems to be close to your other two issues.

**Comment by highness28** [comment]
with my experience with the package you don't have to reconnect again go-whatsapp if ever you closed the phone or the phone loses internet connection. the only problem I've encounter if go-whatsapp disconnect from whatsapp's web socket which is the only time you will be needing to reconnect go-whatsapp. the second problem is that if you connected your phone to another web app like the official web whatsapp. Whatsapp web socket will surely close your socket session to their server.

I have tried @kaxap 's package but somehow it doesn't reconnect itself (the socket).

What I did is to make the variable kaxap made into a exportable variable and access it in my service and when that variable turns to false I will create a new connection and use that connection to synch my .gob file.

the struct is under conn.go on line 79
```
type ConnectionStatus struct {
	sync.RWMutex
	connected bool // I capitalized this in order to use this outside or make it exportable
}
```

the reason I did this is because the Disconnect function doesn't eliminate the resources immediately so I have to find a way to do that.

What my point here is that doing that loop will cause a problem for you handling those resources if ever your phone got disconnected and your service got disconnected also to whatsapp web socket and tries to create the connection again and again.

**Comment by kaxap** [comment]
@highness28 you don't need that to be public in order to access it. There is public Connected() method which appropriately uses related mutex and returns contents of connected bool variable.

Regarding automatic reconnection, I don't remember implementing it. I guess it's has to be handled in app itself, like you did in yours.

**Comment by highness28** [comment]
@kaxap apparently the `Connected()` method of `ConnectionStatus` struct is undefined based on the error

tried doing this

`fmt.Println(wac.ConnectionStatus.Connect())`

where `wac` is my `Conn` struct.

it says

```wac.ConnectionStatus undefined (type *whatsapp.Conn has no field or method ConnectionStatus, but does have whatsapp.connectionStatus)(undefined)```

while doing this

`fmt.Println(wac.connectionStatus)`

gives me an error of

```wac.connectionStatus undefined (cannot refer to unexported field or method connectionStatus)(undefined)```

**Comment by kaxap** [comment]
ah I see, connectionStatus is private, didn't notice that, my bad.

So there are `wac.isConnected()` and `wac.connectionStatus.Connected()`. This is quite confusing, I admit.
The problem is in the naming, since the methods actually have different meaning: 

`wac.isConnected()` turns to false whenever connection to the server is lost either by read/write error or calling `reconnect()` method which closes the current connection. However, it turns to true whenever `sendKeepAlive` is called. Additionally, `sendKeepAlive` called perpetually in the `keepAlive` goroutine so `wac.isConnected()` can return true even though there is no actual connection to the server.

On the other hand `wac.connectionStatus.Connected()` returns true after initial connection has been established. However, once `websocket.CloseNormalClosure` or `websocket.CloseGoingAway` has been encountered in `readPump()` or [socket close handler](https://github.com/kaxap/go-whatsapp/blob/250a33edec6a444a647c037bc7a854f6b65f251f/conn.go#L199) it will return false and never turn to true. 

tl;dr: `wac.isConnected()` false means there is no connection, but reconnection is possible, true basically means nothing; `wac.connectionStatus.Connected()` returns false when the socket was closed normally.

**Comment by SchulteMK** [comment]
I think this issue is resolved due to #114. If not please reopen this issue or create a new one
